### DISCUSSION TITLE
documentation - incorrect names for environment variables for Azure authentication

### DISCUSSION THREAD

**Discussion by cscoon** [tomkerkhove/promitor#1494] [Q&A] [answered]
**Report**

The documentation located at: https://promitor.io/configuration/v2.x/azure-authentication.html incorrectly states that the environment variables holding the app ID and app key for authentication into Azure are PROMITOR_AUTH_APPID and PROMITOR_AUTH_APPKEY.

The correct values are AUTH_APPID and AUTH_APPKEY.  For reference: https://github.com/tomkerkhove/promitor/blob/master/src/Promitor.Core/EnvironmentVariables.cs

**Expected Behavior**

I expected the environment variables I created per documentation to prevent an exception from being thrown.

**Actual Behavior**

An exception was thrown.  (Sorry... but the "expected" and "actual" are required fields! :) )

**Steps to Reproduce the Problem**

See references above.

**Component**

Resource Discovery

**Version**

2.0.0

**Configuration**

_No response_

**Logs**

_No response_

**Platform**

Other

**Contact Details**

_No response_

asdf

**Comment by tomkerkhove** [comment]
The code you are referring to does not include the prefix.

Did you have issues at runtime? If so, can you share them please?

**Comment by cscoon** [comment, marked as answer]
Correct.  The code does NOT include the prefix.  The documentation in the link I provided DOES include the prefix.  

So the documented name of the environment variables does not match what the code is actually looking for.

**Reply by tomkerkhove** [reply]
It does match because the runtime has a prefix configured (https://github.com/tomkerkhove/promitor/blob/master/src/Promitor.Agents.ResourceDiscovery/Program.cs#L93), that's why I'm asking if you are having issues or not as you should not have them.

**Reply by cscoon** [reply]
Ah.... you are correct.  I ran into the need for environment variables when I didn't have the variables defined on my machine, and so I found them in source code.  Later, I found the documentation and noticed the difference in the names.

I just tried running the service locally and it started correctly with the Promitor_ prefix, so this issue can be closed.  My apologies for the false alarm!

**Reply by tomkerkhove** [reply]
No problem at all, thanks for reporting!
